### ISSUE TITLE
rxDataObjOpen unit test randomly fails on ctime/mtime assertion

### ISSUE THREAD

**Issue by korydraughn** [irods/irods#5863]
- [x] master
- [x] 4-2-stable
---
### iRODS Version, OS and Version
Tip of master and 4-2-stable

### What did you try to do?
I ran the unit tests.

### Expected behavior
I expected all tests to pass consistently.

### Observed behavior (including steps to reproduce, if applicable)
The following line fails sometimes. If you run the specific test case that contains this line multiple times, you will eventually see the failure. It appears the server may be using different timestamps for the ctime/mtime when creating a replica.
https://github.com/irods/irods/blob/205ee0e3bd7e3b4107ed44c8c49027d4ec21b2f9/unit_tests/src/test_rc_data_obj.cpp#L742

**Comment by alanking** [comment]
This is being caused by the finalize logic of `rsDataObjRepl` always setting the mtime for the new replica to the current time which can sometimes differ from the create time. This is not the intended behavior. For comparison, `rsDataObjClose` never sets the mtime for new replicas. This should probably be the case for replication as well.

**Comment by alanking** [comment]
Actually, when I said "This is not the intended behavior.", I meant to say that this is a departure from the historical behavior. I think it makes the most sense that the create time and mtime match from the time of creation for a new replica.

To illustrate the comparison I mentioned, here is the logic in `rsDataObjClose` for finalizing an ordinary open-for-write/close: https://github.com/irods/irods/blob/fdb138a6c2d7025f8de0af0308229b3c21d2b71a/server/api/src/rsDataObjClose.cpp#L367-L386
The mtime is only updated when opened for write and when bytes are written, not on create.

Meanwhile, in `rsDataObjRepl`... https://github.com/irods/irods/blob/fdb138a6c2d7025f8de0af0308229b3c21d2b71a/server/api/src/rsDataObjRepl.cpp#L235-L239
The mtime is updated no matter what when finalizing the replica.

**Comment by korydraughn** [comment]
Yes. It makes sense for the mtime to match the ctime for new replicas. That is my first thought. However ...

What if the mtime of the new replica matched the source replica's mtime (for replication)? Is there benefit in that? Having the mtime match the source replica's mtime would introduce the case where the ctime is later than the mtime. Would users care about that?

**Comment by alanking** [comment]
After some discussion, we have decided that the mtime and create time should be individual to the replicas (that is, neither will be inherited from the source replica).

As far as whether the create time and mtime should be equal on create, we decided that we should try to emulate UNIX-like systems for updating the mtimes. It seems like the mtime should be updated any time a write occurs: https://stackoverflow.com/questions/15183067/is-mtime-the-begin-time-or-the-end-time-of-the-modification-process

If we wanted to model that behavior, it would be a departure from the historical behavior where create time and mtime are equal. The create time would then become more analogous with the `crtime` for an inode in Linux (https://www.kernel.org/doc/Documentation/filesystems/ext4/inodes.rst):
